### PR TITLE
Wip

### PR DESCRIPTION
**Pull request by raicho81**
Initial definitions of the library etc.

### CODE REVIEW

File: raft_consensus/protocol.nim [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+                        #                                       #
+                        #   RAFT Messages Protocol definition   #
+                        #                                       #
+import types
+
+type
+    # RAFT Node Messages definitions
+    RAFTMessageOps* = enum
+        REQUEST_VOTE        = 0,
```
Comment (zah) [inline raft_consensus/protocol.nim]: Please re-format these files to conform to the EditorConfig configuration present in this repo (you must use 2 spaces for indentation).

File: raft_consensus.nim [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# nim-raft-consesnsus
```
Comment (zah) [inline raft_consensus.nim]: Let's remote the `odt` roadmap file above in favor of a markdown file with the same content.

File: raft_consensus/types.nim [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    RAFTNode* = object
+        # Timers
+        voting_timout: uint64
+        heart_beat_timeout: uint64
+        # etc. timers
+
+        # Mtx definitions go here
+        raft_state_mutex: Lock
+        raft_log_mutex: Lock
+        raft_comm_mutex_receive_msg: Lock
```
Comment (zah) [inline raft_consensus/types.nim]: Let's follow the [NEP1 naming conventions](https://nim-lang.org/1.0.0/nep1.html#introduction-naming-conventions) (i.e. use `camelCase` for field names).

File: raft_consensus.nim (hunk 2 above)
Comment (raicho81) [reply raft_consensus.nim]: Done.

File: raft_consensus/types.nim (hunk 3 above)
Comment (raicho81) [reply raft_consensus/types.nim]: Got it.

File: raft_consensus/protocol.nim (hunk 1 above)
Comment (raicho81) [reply raft_consensus/protocol.nim]: Got it.

File: LICENSE-APACHEv2 [hunk 4]
```diff
@@ -1,5 +1,5 @@
-The Nimbus GUI is licensed under the Apache License version 2
-Copyright (c) 2023 Status Research & Development GmbH
+nim-raft is licensed under the Apache License version 2
+Copyright (c) 2018 Status Research & Development GmbH
```
Comment (zah) [inline LICENSE-APACHEv2]:
```suggestion
Copyright (c) 2023 Status Research & Development GmbH
```

File: LICENSE-MIT [hunk 5]
```diff
@@ -1,5 +1,5 @@
-The Nimbus GUI is licensed under the MIT License
-Copyright (c) 2023 Status Research & Development GmbH
+nim-raft is licensed under the MIT License
+Copyright (c) 2018 Status Research & Development GmbH
```
Comment (zah) [inline LICENSE-MIT]:
```suggestion
Copyright (c) 2023 Status Research & Development GmbH
```

**Pull request merged**